### PR TITLE
Clarify the license

### PR DESCRIPTION
**Pull request by Adda0**
This PR applies some additional improvements to the licence terms, as per the suggestion of @kilohsakul.

### CODE REVIEW

**Review by jurajsic — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -88,9 +88,9 @@ Tests for Z3-Noodler are located in [src/test/noodler](src/test/noodler).
 
 ## Licensing
 
-The code of Z3-Noodler is distributed under the MIT License. See [LICENSE.md](./LICENSE.md).
+Z3-Noodler is distributed under the MIT License. See [LICENSE.md](./LICENSE.md).
```
Comment (kilohsakul) [inline README.md:91]: "licensed" may be better than "distributed", but it does not matter so much

File: LICENSE.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Z3-Noodler
 
-This project is an SMT string solver, a modified version of Z3 SMT Solver, distributed under the MIT License.
+This project, Z3-Noodler, is an SMT string solver, distributed under the MIT License.
```
[the remark this answers is not in the document]
Comment (Adda0) [reply LICENSE.md:3]: Fixed.

File: README.md (hunk 1 above)
Comment (Adda0) [reply README.md:91]: Distributed is the standard legalise for this thing. `Licensed` can also be used.

File: README.md [hunk 3]
```diff
@@ -88,9 +88,9 @@ Tests for Z3-Noodler are located in [src/test/noodler](src/test/noodler).
...
 
-Z3-Noodler is a derivative work of Z3, distributed under the MIT License. See [LICENSE_Z3.txt](./LICENSE_Z3.txt).
+Z3-Noodler is a derivative work of Z3 SMT solver, distributed under the MIT License. See [LICENSE_Z3.txt](./LICENSE_Z3.txt).
```
Comment (kilohsakul) [inline README.md:93]: Z3-Noodler is a derivative of the SMT solver Z3. Z3-Noodler is distributed under the MIT License. 

(don't like "Z3 SMT solver", ugly english, and it was not clear what the second sentence was refering to, Z3 ro Z3-Noodler)

Comment (Adda0) [reply README.md:93]: ` Z3 SMT solver` is the official name from Z3 materials.

Comment (Adda0) [reply README.md:93]: And this is supposed to mention the Z3 original licence, not the Z3-Noodler license.

File: LICENSE.md [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 
-The original Z3 code from the [Z3 repository](https://github.com/Z3Prover/z3) is distributed under the MIT License in [LICENSE_Z3.txt](./LICENSE_Z3.txt).
+Z3-Noodler is a modified version of Z3 SMT solver.
```
Comment (kilohsakul) [inline LICENSE.md:5]: "the SMT solver Z3"

Comment (Adda0) [reply LICENSE.md:5]: Fixed.

File: README.md (hunk 3 above)
Comment (Adda0) [reply README.md:93]: Fixed.

File: README.md (hunk 1 above)
Comment (Adda0) [reply README.md:91]: Fixed.

**Pull request merged**